src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -94,8 +118,49 @@ public VisualStudioProjectTracker(IServiceProvider serviceProvider)
             }
         }
 
+        private void WaitForScheduledForegroundAffinitizedActions()
+        {
+            AssertIsForeground();
+
+            lock (_gate)
+            {
+                _taskForForegroundAffinitizedActions.Wait();
```

Comment:
Lets say, I was as much confused as skeptical. If it turns out we need to force complete pending tasks, we will find out later and then fix it.